File: src/entity/panier.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity;

/**
 *
 * @author mikhail
 */
public class panier {
    private int id_panier ; 
    private int id_user ; 
    private int id_produit ; 
    private String nom;
    private int prix;
    private int qty;

    public panier(int id_user, int id_produit, String nom, int prix, int qty) {
        this.id_user = id_user;
        this.id_produit = id_produit;
        this.nom = nom;
        this.prix = prix;
        this.qty = qty;
    }

    public panier(String nom, int prix, int qty) {
        this.nom = nom;
        this.prix = prix;
        this.qty = qty;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public int getPrix() {
        return prix;
    }

    public void setPrix(int prix) {
        this.prix = prix;
    }

    public panier(int id_user, int id_produit, int qty) {
        this.id_user = id_user;
        this.id_produit = id_produit;
        this.qty = qty;
    }

    public int getQty() {
        return qty;
    }

    public void setQty(int qty) {
        this.qty = qty;
    }

    public panier() {
    }

    public panier(int id_user, int id_produit) {
        this.id_user = id_user;
        this.id_produit = id_produit;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }

    public int getId_produit() {
        return id_produit;
    }

    public void setId_produit(int id_produit) {
        this.id_produit = id_produit;
    }

    @Override
    public int hashCode() {
        int hash = 5;
        hash = 47 * hash + this.id_user;
        hash = 47 * hash + this.id_produit;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final panier other = (panier) obj;
        if (this.id_user != other.id_user) {
            return false;
        }
        if (this.id_produit != other.id_produit) {
            return false;
        }
        return true;
    }

    public panier(int id_panier, String nom, int prix, int qty) {
        this.id_panier = id_panier;
        this.nom = nom;
        this.prix = prix;
        this.qty = qty;
    }

    public int getId_panier() {
        return id_panier;
    }

    @Override
    public String toString() {
        return  "nom=" + nom + ", prix=" + prix + ", qty=" + qty ;
    }
    
    
    
}

File: Pidev2-3A-user (4)/Pidev2-3A-user/Pidev2-3A-user/src/Entity/Panier.php
<?php

namespace App\Entity;
use Symfony\Component\Validator\Constraints as Assert;

use Doctrine\ORM\Mapping as ORM;

/**
 * Panier
 *
 * @ORM\Table(name="panier", indexes={@ORM\Index(name="id_produit", columns={"id_produit"}), @ORM\Index(name="idUser", columns={"idUser"})})
 * @ORM\Entity
 */
class Panier
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_panier", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idPanier;

    /**
     * @var string
     *
     * @ORM\Column(name="nom", type="string", length=250, nullable=false)
     */
    #[Assert\NotBlank (message:"veuillez saisir  nom ")]

    private $nom;
      /**
     * 
     * @ORM\Column(length=1000)
     * 
     */
    #[Assert\NotBlank (message:"veuillez saisir l'image du l'evenement ")]
    private $image;

    /**
     * @var int
     *
     * @ORM\Column(name="prix", type="integer", nullable=false)
     */
    #[Assert\NotBlank (message:"veuillez saisir prix ")]

    private $prix;

    /**
     * @var int
     *
     * @ORM\Column(name="qty", type="integer", nullable=false)
     */
    #[Assert\NotBlank (message:"veuillez saisir quantité ")]

    private $qty;

    /**
     * @var \Trotinette
     *
     * @ORM\ManyToOne(targetEntity="Trotinette")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_produit", referencedColumnName="id_trot")
     * })
     */
    #[Assert\NotBlank (message:"veuillez saisir id trorinnete ")]

    private $idProduit;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="IdUser", referencedColumnName="IdUser")
     * })
     */
    #[Assert\NotBlank (message:"veuillez saisir id user ")]

    private $idUser;

    public function getIdPanier(): ?int
    {
        return $this->idPanier;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function setImage( $image)
    {
        $this->image = $image;

        return $this;
    }
    

    public function getPrix(): ?int
    {
        return $this->prix;
    }

    public function setPrix(int $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getQty(): ?int
    {
        return $this->qty;
    }

    public function setQty(int $qty): self
    {
        $this->qty = $qty;

        return $this;
    }

    public function getIdProduit(): ?Trotinette
    {
        return $this->idProduit;
    }

    public function setIdProduit(?Trotinette $idProduit): self
    {
        $this->idProduit = $idProduit;

        return $this;
    }

    public function getIdUser(): ?User
    {
        return $this->idUser;
    }

    public function setIdUser(?User $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }

    
}
